src/_pytest/python.py

```diff
@@ -504,14 +504,26 @@ def _genfunctions(self, name: str, funcobj) -> Iterator["Function"]:
 
             for callspec in metafunc._calls:
                 subname = f"{name}[{callspec.id}]"
-                yield Function.from_parent(
+                node = Function.from_parent(
                     self,
                     name=subname,
                     callspec=callspec,
                     fixtureinfo=fixtureinfo,
                     keywords={callspec.id: True},
                     originalname=name,
                 )
+                # if usefixtures is added via a parameter, then there will be
+                # fixtures missing from the node.fixturenames
+                callspec_usefixtures = tuple(
+                    arg
+                    for mark in node.iter_markers(name="usefixtures")
+                    for arg in mark.args
+                    if arg not in node.fixturenames
+                )
+                if callspec_usefixtures:
+                    # node.fixturenames must be unique for this parameter
```

Comment:
can you explain why the default `usefixtures` implementation  in `getfixtureinfo()` doesn't work ? 

I would think this should be fixed in `Function.__init__` so it's one code path that does the logic of expending the fixtures names out

Reply:
I think fixtureinfo is shared between multiple generate functions, and it should be a clean copy for each.

Reply:
and that why your test case is adding it on the 2nd parametrize

Reply:
> can you explain why the default usefixtures implementation in getfixtureinfo() doesn't work ?

Because getfixtureinfo is called only for the base function, for parameterized function calls fixtureinfo is assigned from base function without updates

> I would think this should be fixed in Function.__init__ so it's one code path that does the logic of expending the fixtures names out
> I think fixtureinfo is shared between multiple generate functions, and it should be a clean copy for each.

A copy should only be for an instance with a unique set of fixtures.

Updated the solution with the fixture update in __init__.